packages/vue-components/src/Submenu.vue

```diff
@@ -0,0 +1,177 @@
+  <li ref="submenu" :class="classes">
+    <slot name="button">
+      <a
+        class="submenu-toggle"
+        role="button"
+        :class="{disabled: disabled}"
+      >
+        <slot name="_header">
+          <slot name="header"></slot>
+        </slot>
+      </a>
+    </slot>
+    <slot name="dropdown-menu">
+      <ul class="dropdown-menu">
+        <slot></slot>
+      </ul>
+    </slot>
+  </li>
+</template>
+
+<script>
+import { toBoolean } from './utils/utils';
+import $ from './utils/NodeList';
+import positionSubmenu from './utils/submenu';
+
+export default {
+  props: {
+    show: {
+      type: Boolean,
+      default: false,
+    },
+    'class': null,
+    addClass: {
+      type: String,
+      default: '',
+    },
+  },
+  data() {
+    return {
+      dropright: true,
+      dropleft: false,
+    };
+  },
+  computed: {
+    classes() {
+      return [
+        this.class, this.addClass, 'dropdown-submenu',
+        { 'dropright': this.dropright, 'dropleft': this.dropleft },
+      ];
+    },
+    disabledBool() {
+      return toBoolean(this.disabled);
+    },
+    showBool() {
+      return toBoolean(this.show);
+    },
+    slots() {
+      return this.$slots.default;
+    },
+  },
+  methods: {
+    hideSubmenu() {
+      this.show = false;
+      $(this.$refs.submenu).findChildren('ul').each(ul => ul.classList.toggle('show', false));
+      this.alignMenuRight();
+    },
+    showSubmenu() {
+      this.show = true;
+      $(this.$refs.submenu).findChildren('ul').each((ul) => {
+        ul.classList.toggle('show', true);
+        if (positionSubmenu.isRightAlign(ul)) {
+          this.alignMenuRight();
+        } else {
+          this.alignMenuLeft();
+        }
+        positionSubmenu.preventOverflow(ul);
+      });
+    },
+    alignMenuRight() {
+      this.dropright = true;
+      this.dropleft = false;
+    },
+    alignMenuLeft() {
+      this.dropright = false;
+      this.dropleft = true;
+    },
+  },
+  created() {
+    this._submenu = true;
+  },
+  mounted() {
+    const $el = $(this.$refs.submenu);
+    if (this.show) {
+      this.showSubmenu();
+    }
+    $el.onBlur(() => { this.hideSubmenu(); }, false);
+    $el.findChildren('a,button').on('click', (e) => {
+      e.preventDefault();
+      if (this.disabledBool) { return false; }
+      if (this.showBool) {
+        this.hideSubmenu();
+      } else {
+        this.showSubmenu();
+      }
+      return false;
+    });
+    $el.findChildren('a,button').on('mouseover', (e) => {
+      e.preventDefault();
+      if (window.innerWidth > 768) {
+        e.target.click();
+      }
+    });
+  },
+  beforeDestroy() {
+    const $el = $(this.$refs.submenu);
+    $el.offBlur();
+    $el.findChildren('a,button').off();
+    $el.findChildren('ul').off();
+  },
+};
+</script>
+
+<style scoped>
+.dropdown-submenu {
+  color: #212529 !important;
+  padding: 0 !important;
+  position: relative;
+}
+
+@media (max-width: 767px) {
+  .dropdown-submenu > ul {
```

Comment:
hmm can this be moved to the same media query below?

Reply:
Thanks! Updated :)